### PR TITLE
transitioning title attribute to render as tooltip, and adding readon…

### PR DESCRIPTION
**Pull request by petdun2519**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000

#### Description of changes

Title attribute will now be rendered as tooltip, readonly state provided for command bar items

#### Focus areas to test

(optional)

### CODE REVIEW

**Comment by micahgodbolt** [comment]
This is basically adding a "readOnly" mode to contextual menu items (and should therefore also be added to buttons). It addresses the idea that a button can be focusable, but not actionable (as opposed to disabled which is treated as not really even there). It would probably have the same styles as disabled (though some design review would probably be useful). @joschect, @betrue-final-final : Has this approach been talk about/mocked up recently?

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx [hunk 1]
```diff
@@ -282,6 +296,13 @@ export class CommandBar extends BaseComponent<ICommandBarProps, ICommandBarState
         </TooltipHost>
       );
     }
+    else if (tooltipContent) {
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: Nit: this should be on the same line as the closing {

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx [hunk 2]
```diff
@@ -399,6 +420,10 @@ export class CommandBar extends BaseComponent<ICommandBarProps, ICommandBarState
 
   private _onItemClick(item: IContextualMenuItem): (ev: React.MouseEvent<HTMLButtonElement>) => void {
     return (ev: React.MouseEvent<HTMLButtonElement>): void => {
+      if (item.readOnly) {
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: If this is added to command bar, it should also be added to the contextualmenu click handler.

**Comment by joschect** [comment]
@micahgodbolt  I'm a little hesitant about readonly being applied to non-input fields. Read only already has some connotation that the content is editable. While in the case of buttons/clickable things you can't edit those.

**Comment by micahgodbolt** [comment]
@joschect good point, readonly does already have a meaning in HTML. How about we just call it smoosh?

j/k

Regardless of the name, I think the state makes sense.

**Comment by petdun2519** [comment]
@micahgodbolt so you want me to add this behavior to the button components also? @joschect how about, 'actionable'? because disabled is already taken unfortunately

**Comment by micahgodbolt** [comment]
we usually want a Boolean prop to be false by default, so that changing it to true is the exception. I asked around on term ideas and one of my favorites was 'inactive'. Another was 'suppressed', or 'locked' (though locked seems like a specific reason for being inactive or suppressed).

**Comment by petdun2519** [comment]
@micahgodbolt ok I will change this to inactive. But do I need to implement this in button or can it happen in a subsequent pr?

**Comment by micahgodbolt** [comment]
If this is blocking you, then doing button later is okay. But that is technical debt that has to be cleaned up before you're able to move to the 6.0 CommandBar.

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx (hunk 2 above)
Comment (petdun2519) [reply packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: @joschect I added a comment on this prop that it's implementation is not supported by all components, like for contextualmenu and for button as micah as mentioned, then other teams can come in and add the functionality on these components as they need them. is that ok?

**Comment by petdun2519** [comment]
@joschect I have made the edits you suggested

File: packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx [hunk 3]
```diff
@@ -188,8 +188,16 @@ export class CommandBar extends BaseComponent<ICommandBarProps, ICommandBarState
...
       isLink ? ('ms-CommandBarItem-link ' + styles.itemLink) : ('ms-CommandBarItem-text ' + styles.itemText),
       !item.name && ('ms-CommandBarItem--noName ' + styles.itemLinkIsNoName),
-      (expandedMenuItemKey === item.key) && ('is-expanded ' + styles.itemLinkIsExpanded)
+      (expandedMenuItemKey === item.key) && ('is-expanded ' + styles.itemLinkIsExpanded),
+      item.inactive ? styles.inactive : ''
     );
+
+    let tooltipContent = '';
+
+    if (item.title) {
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: Doesn't this add a tooltip to every command bar item? Is that intended?

Comment (petdun2519) [reply packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: @micahgodbolt only if they provide a title attribute

Comment (micahgodbolt) [reply packages/office-ui-fabric-react/src/components/CommandBar/CommandBar.tsx]: ah title attribute, yup. gotcha. was thinking title was the text in the button.

**Pull request merged**